### PR TITLE
Feature/15 add gh action to execute python tests

### LINKED ISSUE #15: Add GH Action to execute Python tests
Add GitHub Action that executes all Python test cases upon push to branch. This should be pretty simple.

Rough sketch:
- https://stackoverflow.com/q/60298449/104274

Docs:
- https://docs.github.com/en/actions

### PR DESCRIPTION
**Pull request by BoxG11**
Added a github action to test all files.

### CODE REVIEW

File: .github/workflows/pytest.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v2
+      with:
+        python-version: "3.10"
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install flake8 pytest
```
Comment (bkircher) [inline .github/workflows/pytest.yml]: Please remove this line. pytest and things is installed via dev-requirements.txt

File: .github/workflows/pytest.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        if [ -f dev-requirements.txt ]; then pip install -r dev-requirements.txt; fi
```
Comment (bkircher) [inline .github/workflows/pytest.yml]: You can remove the test here if dev-requirements.txt exists. Just assume it does :smile: If it doesn't this pipeline fails, which is good.

**Review by bkircher — CHANGES_REQUESTED** [review]
Cool, thanks for working on this. Can you take a look at two requested changes below?

As a side note: If you run flake8 or pylint on your terminal in this project, you see that there is no proper config for either of these tools and there are lots of warnings. I think it is a good next thing to work on, isn't it? :smile:  I will create an issue for this.

As we can see, there is still a Travis CI job executed, besides the GH Action. I forgot about this. Let's just ignore this for now :sweat_smile: 

The Travis CI thing fails when it tries to push a package to PyPI because of missing authentication tokens. Totally reasonable, since we don't want to push to PyPI from PR branches. I think, in the mid-term we should get rid of Travis CI and only use GH Actions. @Aldemuro @begotxe any thoughts on this?

**Comment by bkircher** [comment]
Created a new issue for that linter stuff: https://github.com/gridscale/gridscale_api_client_python/issues/17

**Comment by BoxG11** [comment]
The requested changes have been made. Check here: [https://github.com/gridscale/gridscale_api_client_python/commit/67fdc34a76f69e56dba7e7da93c740a79ca80946#](url)

**Review by bkircher — APPROVED** [review]

**Comment by bkircher** [comment]
Thanks! Merged

**Pull request merged**